zeebe/exporters/camunda-exporter/src/main/java/io/camunda/exporter/tasks/archiver/OpenSearchArchiverRepository.java

```diff
@@ -113,39 +118,29 @@ public CompletableFuture<ArchiveBatch> getBatchOperationsNextBatch() {
   }
 
   @Override
-  public CompletableFuture<Void> setIndexLifeCycle(final String destinationIndexName) {
+  public CompletableFuture<Void> setIndexLifeCycle(final String... destinationIndexName) {
     if (!retention.isEnabled()) {
       return CompletableFuture.completedFuture(null);
     }
-
-    final AddPolicyRequestBody value = new AddPolicyRequestBody(retention.getPolicyName());
-    final var request =
-        Requests.builder().method("POST").endpoint("_plugins/_ism/add/" + destinationIndexName);
-
-    return sendRequestAsync(
-            () ->
-                genericClient.executeAsync(
-                    request.json(value, genericClient._transport().jsonpMapper()).build()))
-        .thenComposeAsync(
-            response -> {
-              if (response.getStatus() >= 400) {
-                return CompletableFuture.failedFuture(
-                    new ExporterException(
-                        "Failed to set index lifecycle policy for index: "
-                            + destinationIndexName
-                            + ".\n"
-                            + "Status: "
-                            + response.getStatus()
-                            + ", Reason: "
-                            + response.getReason()));
-              }
-              return CompletableFuture.completedFuture(null);
-            });
+    return CompletableFuture.allOf(
+        Arrays.stream(destinationIndexName)
+            .map(this::applyPolicyToIndex)
+            .toArray(CompletableFuture[]::new));
   }
 
   @Override
   public CompletableFuture<Void> setLifeCycleToAllIndexes() {
-    return CompletableFuture.completedFuture(null);
+    if (!retention.isEnabled()) {
+      return CompletableFuture.completedFuture(null);
+    }
+    final String indexWildCard = "^" + indexPrefix + INDEX_WILDCARD;
+
+    try {
+      return fetchIndexMatchingIndexes(indexWildCard)
+          .thenCompose(indices -> setIndexLifeCycle(indices.toArray(String[]::new)));
```

Comment:
:x: Use `thenComposeAsync` and specify the executor. This ensures that we will skip the callback if the background task manager is closed.